Mathlib/Analysis/InnerProductSpace/PiL2.lean

```diff
@@ -422,6 +422,15 @@ lemma enorm_eq_one (b : OrthonormalBasis ι 𝕜 E) (i : ι) :
 lemma inner_eq_zero (b : OrthonormalBasis ι 𝕜 E) {i j : ι} (hij : i ≠ j) :
     ⟪b i, b j⟫ = 0 := b.orthonormal.inner_eq_zero hij
 
+lemma inner_eq [DecidableEq ι] (b : OrthonormalBasis ι 𝕜 E) {i j : ι} :
+    ⟪b i, b j⟫ = if i = j then 1 else 0 := by
+  by_cases h : i = j
+  · simp only [h, ↓reduceIte]
+    apply RCLike.ext
+    · simp [inner_self_eq_norm_sq]
+    · simp
```

Comment:
I think this should be an `inner_eq_one` simp lemma.

Reply:
You can also simplify to `simp [h, inner_self_eq_norm_sq_to_K]`, but I think the lemma is still worth having.